code/modules/mob/living/living.dm

```diff
@@ -546,67 +558,29 @@ Sorry Giacom. Please don't be mad :(
 		else
 			return 0
 
-	if (restrained())
+	var/atom/movable/pullee = pulling
+	if(pullee && get_dist(src, pullee) > 1)
 		stop_pulling()
-
-
-	var/cuff_dragged = 0
-	if (restrained())
-		for(var/mob/living/M in range(1, src))
-			if (M.pulling == src && !M.incapacitated())
-				cuff_dragged = 1
-	if (!cuff_dragged && pulling && !throwing && (get_dist(src, pulling) <= 1 || pulling.loc == loc))
```

Comment:
we no longer need to check for throwing every we move while pulling something. The throw procs handle this themselves, and you can't start pulling or be pulled mid-flight.